test/test_container_application.py

```diff
@@ -0,0 +1,194 @@
+        ) == 0
+        # test if php.d directory is writeable
+        assert PodmanCLIWrapper.podman_run_command_and_remove(
+            cid_file_name=VARS.IMAGE_NAME,
+            cmd="[ -w \\${PHP_SYSCONF_PATH}/php.d ]",
+            return_output=False
+        ) == 0
+
+
+class TestPHPClearEnvContainer:
+    def setup_method(self):
+        self.s2i_app = build_s2i_app(test_app, container_args="-e PHP_CLEAR_ENV=OFF")
+
+    def teardown_method(self):
+        if self.s2i_app:
+            self.s2i_app.cleanup()
+
+    def test_run_app_test(self):
+        """
+        Test checks if php-version is proper one
+        and specific pages returns expected message or return code.
+        """
+        skip_clear_env_tests()
+        cid_file_name = self.s2i_app.app_name
+        assert self.s2i_app.create_container(cid_file_name=cid_file_name, container_args="--user=100001")
+        assert ContainerImage.wait_for_cid(cid_file_name=cid_file_name)
+        cid = self.s2i_app.get_cid(cid_file_name=cid_file_name)
+        assert cid
+        assert VARS.VERSION in PodmanCLIWrapper.podman_exec_shell_command(
+            cid_file_name=cid,
+            cmd="php --version"
+        )
+        assert VARS.VERSION in PodmanCLIWrapper.podman_exec_shell_command(
+            cid_file_name=cid,
+            cmd="php --version",
+            used_shell="/bin/sh"
+        )
+
+        cip = self.s2i_app.get_cip(cid_file_name=cid_file_name)
+        assert cip
+        # This function tests PHP session
+        assert self.s2i_app.test_response(
+            url=cip, page="/session_test.php",
+            expected_output="Passed"
+        )
+        # Assert tests if connect to http://<ip>:8080 works
+        assert self.s2i_app.test_response(
+            url=cip
+        )
+        # Assert tests if connect to https://<ip>:8443 works
+        assert self.s2i_app.test_response(
+            url=f"https://{cip}", port=8443
+        )
+        # test if configuration is writeable for php.ini
+        assert PodmanCLIWrapper.podman_run_command_and_remove(
+            cid_file_name=VARS.IMAGE_NAME,
+            cmd="[ -w \\${PHP_SYSCONF_PATH}/php.ini ]",
+            return_output=False
+        ) == 0
+        # test if php.d directory is writeable
+        assert PodmanCLIWrapper.podman_run_command_and_remove(
+            cid_file_name=VARS.IMAGE_NAME,
+            cmd="[ -w \\${PHP_SYSCONF_PATH}/php.d ]",
+            return_output=False
+        ) == 0
+        if VARS.OS == "rhel9" or VARS.OS == "rhel10":
+            # Checking if clear_env = no is set in /etc/php-fpm.d/www.conf file.
+            file_content = PodmanCLIWrapper.podman_get_file_content(
+                cid_file_name=cid,
+                filename="/etc/php-fpm.d/www.conf"
+            )
+            found = False
+            for line in file_content.split('\n'):
+                if re.search("^clear_env = no", line):
+                    found = True
+            assert found
```

Comment:
You can use `else` branch for for loop to make it shorter and the possible failure more descriptive.
```suggestion
            for line in file_content.splitlines():
                if re.search("^clear_env = no", line):
                    break
            else:
                assert False, "'clear_env = no' not found in www.conf file."
```

Reply:
Fix I forgot it https://github.com/sclorg/s2i-php-container/pull/501/commits/0bc9061f3d07ae9d1ff744186132ad2977f1335b